handle.go

```diff
@@ -64,38 +64,33 @@ func NewHandle(rules any, keyObfuscatorRegex string, valueObfuscatorRegex string
 	}
 
 	config := newConfig(&encoder.cgoRefs, keyObfuscatorRegex, valueObfuscatorRegex)
-	cRulesetInfo := new(wafRulesetInfo)
+	diagnosticsWafObj := new(wafObject)
 
-	cHandle := wafLib.wafInit(obj, config, cRulesetInfo)
+	cHandle := wafLib.wafInit(obj, config, diagnosticsWafObj)
 	keepAlive(encoder.cgoRefs)
 	// Note that the encoded obj was copied by libddwaf, so we don't need to keep them alive
 	// for the lifetime of the handle (ddwaf API guarantee).
 	if cHandle == 0 {
 		return nil, errors.New("could not instantiate the WAF")
 	}
 
-	defer wafLib.wafRulesetInfoFree(cRulesetInfo)
-
-	errorsMap, err := decodeErrors(&cRulesetInfo.errors)
-	if err != nil { // Something is very wrong
-		return nil, fmt.Errorf("could not decode the WAF ruleset errors: %w", err)
+	diags, err := decodeDiagnostics(diagnosticsWafObj)
+	if err != nil {
+		return nil, fmt.Errorf("could not decode the WAF diagnostics: %w", err)
 	}
 
+	defer wafLib.wafObjectFree(diagnosticsWafObj)
```

Comment:
From the ddwaf_object_free doc, it does not free the top level object. Do we need to free it ?

Reply:
Ok it's irrelevant, the top level object is owned by the go runtime